org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -546,6 +555,90 @@ public static boolean upgradeGradleVersion(String projectUri, String gradleVersi
 		return true;
 	}
 
+	/**
+	 * Get Gradle init script arguments for protobuf support.
+	 */
+	private static List<String> getProtobufInitScriptArgs() {
+		List<String> args = new LinkedList<>();
+		PreferenceManager preferencesManager = JavaLanguageServerPlugin.getPreferencesManager();
+		if (preferencesManager == null) {
+			return args;
+		}
+
+		if (!preferencesManager.getPreferences().isProtobufSupportEnabled()) {
+			return args;
+		}
+
+		File initScript = getProtobufInitScript();
+		if (initScript != null && initScript.exists() && initScript.length() > 0) {
+			args.add("--init-script");
+			args.add(initScript.getAbsolutePath());
+		}
+		return args;
+	}
+
+	/**
+	 * Get the protobuf init script file. By default, the init script will be stored as a sibling file of the
+	 * bundle jar. If any exception happens, a temp file will be created and be used instead.
+	 */
+	private static File getProtobufInitScript() {
+		boolean isDebug = Boolean.parseBoolean(System.getenv("jdt.ls.debug"));
+		// in debug mode, use the resource file directly
+		if (isDebug) {
+			Path path = Paths.get(System.getProperty("user.dir"), "org.eclipse.jdt.ls.core", "gradle", "protobuf", "init.gradle");
+			return path.toFile();
+		}
+
+		// in production mode, extract the init script out of the jar
+		try {
+			CodeSource codeSource = JavaLanguageServerPlugin.class.getProtectionDomain().getCodeSource();
+			if (codeSource == null) {
+				return getProtobufInitScriptTempFile();
+			}
+
+			URL location = codeSource.getLocation();
+			if (location == null) {
+				return getProtobufInitScriptTempFile();
+			}
+
+			URI uri = location.toURI().resolve("init.protobuf.gradle");
+			if (uri == null) {
+				return getProtobufInitScriptTempFile();
+			}
+			File initScript = new File(uri);
+			if (!initScript.exists()) {
+				initScript.createNewFile();
+			}
```

Comment:
I really like this approach actually! 👍

It makes the code much neater. Besides, it will locate to the init script in the code base if I launch JDT-LS independently. And targeting to a location inside the downloaded server folder if I launch from the client. Thus, the debug checking logic can be removed as well.

Thank you for pointing out this. :)